### DISCUSSION TITLE
How to backup and restore the multiple Volume Servers data into separate folders, using weed backup. ex: Vol server1 data on /dir1,Vol sever2 data on /dir2 etc..

### DISCUSSION THREAD

**Discussion by nithinks21** [seaweedfs/seaweedfs#5055] [Q&A]
- **Setup**:
I have a seaweedfs environment with 1 Master, 2 Volume servers, 2 Filers, 2 S3 servers.
 2 Volume servers data are on **voldata_1/dir**  and **voldata_2/dir** directories respectively.

Volume server 1's data
![image](https://github.com/seaweedfs/seaweedfs/assets/86155401/40bd7fd8-36fd-4ce7-8a88-76579e669ea8)

Volume Server 2's data
![image](https://github.com/seaweedfs/seaweedfs/assets/86155401/b324c31e-6bc4-4bb7-878b-8d1403f099ff)

From the data on these folders, It's clear that the data on 
Volume Server 1's data folder has data of volumeIds 2,3,5,6 and 7.
Volume Server 2's data folder has data of volumeIds 1 and 4.

- **Requirement**:
I want to take a back up of the volume servers and restore it back in case of data loss or a crash.

I am running the weed back up command which takes the backup of all the volume Ids in a loop.
**
`weed backup -server=localhost:9333 -dir=./data_volume_weed_backup -volumeId=2`
**

But with the above command, we can only organise the backup data by volumeId's.

On restore, I will need to have this entire data on both the volume servers' data directory.

Instead, Is there a way to take the back up of data from each volume server separately.

**- Question/Issue**:
What is the recommended approach to back up and restore volume server data in case of multiple volume servers?

Thank you,
Nithin

**Comment by chrislusf** [comment]
https://github.com/seaweedfs/seaweedfs/blob/master/weed/command/backup.go this command needs to add the feature you want.

**Comment by pkk-code** [comment]
@chrislusf Would it be possible to add this feature to backup based on volume server.
